### PR TITLE
Adding preferred region option in ctl workload for DR drill

### PR DESCRIPTION
**Pull request by simplynaveen20**
Adding preferred region option in ctl workload for DR drill

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/Configuration.java [hunk 1]
```diff
@@ -91,6 +91,9 @@
     @Parameter(names = "-manageDatabase", description = "Control switch for creating/deleting underlying database resource")
     private boolean manageDatabase = false;
 
+    @Parameter(names = "-preferredRegionsList", description = "Comma separated preferred regions list")
+    private String preferredRegionsList;
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/Configuration.java:95]: this is a useful config, as we are adding a global config to the benchmark tool, could you please pass this config to the other components in the benchmark tool as well?

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/ctl/AsyncCtlWorkload.java [hunk 2]
```diff
@@ -84,9 +87,19 @@
...
 
     public AsyncCtlWorkload(Configuration cfg) {
+        String preferredRegionsList = cfg.getPreferredRegionsList();
+        List<String> preferredRegions = null;
+        if (StringUtils.isNotEmpty(preferredRegionsList)) {
+            String[] preferredArray = preferredRegionsList.split(",");
+            if (preferredArray != null && preferredArray.length > 0) {
+                preferredRegions = new ArrayList<>(Arrays.asList(preferredArray));
+            }
+        }
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/ctl/AsyncCtlWorkload.java:97]: please factor out this code outside of the `AsyncCtlWorkload.java` to be used by all benchmark components.

**Review by moderakh — COMMENTED** [review]
Thanks Naveen. nice addition. one minor comments.

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/Configuration.java [hunk 3]
```diff
@@ -91,6 +91,9 @@
     @Parameter(names = "-manageDatabase", description = "Control switch for creating/deleting underlying database resource")
     private boolean manageDatabase = false;
 
+    @Parameter(names = "-preferredRegionsList", description = "Comma separated preferred regions list")
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/Configuration.java:94]: this is a useful config, as we are adding a global config to the benchmark tool, could you please pass this config to the other components in the benchmark tool as well?

**Review by moderakh — APPROVED** [review]
LGTM. Thanks Naveen. 

two minor comments:

1. this is a useful config, as we are adding a global config to the benchmark tool, could you please pass this config to the other components in the benchmark tool as well? as we are adding the config to the main configuration we should support it everywhere. otherwise will be confusing when we use the non ctl workload

2. as a side note, Configuration.java should hide all the config parsing capabilities.
see 3.1. Custom types - Single value section from the following list for how to pass a list and parse a list.
https://jcommander.org/

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/Configuration.java (hunk 1 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/Configuration.java]: Thanks for suggestion ,added the string to list converter

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/Configuration.java (hunk 3 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/Configuration.java]: I was debating it originally , now you mentioned that i added for all the workload. Thanks !

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java [hunk 4]
```diff
@@ -181,7 +181,7 @@ protected void onSuccess() {
     protected void onError(Throwable throwable) {
     }
 
-    private void createClients() {
+    private void createClients(Configuration cfg) {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java:184]: We already have the `configuration` object in this class, why do we need to create a separate one ?

Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java]: It was the same object , however redundant reference , removed it now. Thanks

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**